rustual-boy-core/src/v810.rs

```diff
@@ -265,6 +295,26 @@ impl V810 {
     pub fn step(&mut self, interconnect: &mut Interconnect) -> (usize, bool) {
         let original_pc = self.reg_pc;
 
+        #[cfg(not(feature = "cpu-trace"))]
+        macro_rules! trace_add {
+            ($($arg:expr),*) => {};
+        }
+
+        #[cfg(feature = "cpu-trace")]
+        macro_rules! trace_add {
+            ($te:expr) => {
+                if self.trace_enabled {
+                    let trace_entry = $te;
```

Comment:
I did that because rust wont allow self.trace_entries.push($var), you get these barrow errors.

```
error[E0502]: cannot borrow `*self` as immutable because `self.trace_entries` is also borrowed as mutable
    |
308 |                     self.trace_entries.push($te);
    |                     ------------------         - mutable borrow ends here
    |                     |
    |                     mutable borrow occurs here
...
315 |         trace_add!(TraceEntry::Pc(self.reg_pc()));
    |         --------------------------^^^^------------
    |         |                         |
    |         |                         immutable borrow occurs here
    |         in this macro invocation
```